### ISSUE TITLE
Volar packages renamed

### ISSUE THREAD

**Issue by tobiasdiez** [simonhaenisch/prettier-plugin-organize-imports#98]
The volar packages have been renamed in https://github.com/vuejs/language-tools/pull/3134. It would be nice if the new vue/typescript package could be supported as well.

**Comment by simonhaenisch** [comment]
I haven't followed up at all with the changes in Volar. Just tried and there seem to be breaking changes at least to the exported types, e.g. I was using

```ts
import('@volar/vue-language-core/out/types').LanguageServiceHost
```

in `@volar/vue-typescript` version 1.0.9 but that type doesn't exist anymore in the latest version (or the renamed package).

Some help with this issue would be highly appreciated.

**Comment by tobiasdiez** [comment]
Yes, this interface has been removed: https://github.com/vuejs/language-tools/blob/a73e249f6b5638e7029c0c929eb704e2230a5af3/CHANGELOG.md?plain=1#L157
`createLanguageService` now creates the language host itself and has a different method signature too: https://github.com/vuejs/language-tools/blob/a73e249f6b5638e7029c0c929eb704e2230a5af3/packages/vue-typescript/src/index.ts#L7-L11

**Comment by simonhaenisch** [comment]
Yup that's where I got stuck actually, cause it said sth like "requires 4 arguments but only provided 1" and I didn't really have time to figure out what to pass there.

Looking at the interface again now it looks like instead of a `getVueCompilerOptions` method on the host, you pass them separately when creating the language service. Maybe I'll have some time for this soon, but it'll require a breaking change on my side from what i can tell (not sure why the language-tools didn't have a major bump).

**Comment by Shinigami92** [comment]
@johnsoncodehk Help!

**Comment by johnsoncodehk** [comment]
I don't think there is a need to support the new `@vue/typescript`, we are planning to [deprecate it in 2.0](https://github.com/vuejs/language-tools/issues/3402). Keep using `@volar/vue-typescript` 1.6 would be fine for now.

On the other hand, `@volar/typescript` (the base package of `@vue/typescript`) now has a `decorateLanguageService()` API, which can add Vue support to the existing TS language service instance instead of using `createLanguageService()` to create a new one. `decorateLanguageService()` was originally a 2.0 API, but [it has been released early](https://github.com/vuejs/language-tools/issues/3294). We can check how to integrate `decorateLanguageService()` in this repo after 2.0 if needed.

**Comment by Shinigami92** [comment]
@johnsoncodehk Now Vue 3.4 came out and it looks like `@volar/typescript v1.6.5` is missing support for `defineModel`. So I cant upgrade to `@vue/typescript v1.8.0` due to be stuck with `prettier-plugin-organize-imports`

Does your previous statement with `2.0` still count? What is the progress-state?
Can I workaround it for now somehow? Or can I not use the newest Vue features anymore, because I'm stuck in dependencies?

**Comment by Shinigami92** [comment]
With [Volar v2.0](https://github.com/volarjs/volar.js/releases/tag/v2.0.0), is this the day we can expect new support for it?

**Comment by simonhaenisch** [comment]
I don't work with Vue/Volar at all, so if someone could update me about the current state and ideas how to support it, I'm all ears (:

**Comment by so1ve** [comment]
@simonhaenisch What do you need? :)

**Comment by simonhaenisch** [comment]
Details about how to technically do this with Volar 2:

https://github.com/simonhaenisch/prettier-plugin-organize-imports/blob/9609e4e2b9be2df202edd522b071e2c699348ba3/lib/get-language-service.js#L14

https://github.com/simonhaenisch/prettier-plugin-organize-imports/blob/9609e4e2b9be2df202edd522b071e2c699348ba3/lib/service-host.js#L49-L65

**Comment by arkandias** [comment]
Hi guys, can't do much to help here, just to let you know that your work fixing this issue is greatly appreciated!

**Comment by mreduar** [comment]
Does anyone know of an alternative that works with the latest vue updates?

**Comment by ddzy** [comment]
Guys, any solutions?

**Comment by mreduar** [comment]
> Guys, any solutions?

I removed `@volar/vue-typescript` and `@vue/typescript`, and it started working fine.

Also as @johnsoncodehk said at https://github.com/vuejs/language-tools/issues/3402 

> @vue/typescript is an over-encapsulation, we can use @volar/typescript directly.

Besides it is already deprecated in V2. Installing `@volar/typescript` also works fine.

**Comment by Shinigami92** [comment]
I wont like to depend on deprecated packages, cause I wont get updated then with new features related to the replacement of the deprecated package.  
Also there is also the VSCode extension that relies on the vue and typescript language servers and I don't want to either have multiple language servers running resulting in potential conflicts nor I wont like to be unable to benefit from newest extension feature relying of new dependencies.

If a package is marked as deprecated, there is a reason for that and it should be worked on.
If @simonhaenisch does not have the time and capacity, maybe there should be an additional maintainer with approval, merge and publish rights, so we can move forward.

**Comment by johnsoncodehk** [comment]
This is a PR to migrate to @volar/typescript+@vue/language-core: #129

**Comment by simonhaenisch** [comment]
Published as `4.0.0` 🎉

**Comment by mreduar** [comment]
> Published as `4.0.0` 🎉
> 
> Please try it out and let me know if there are any issues 🙏

Is it still necessary to install `@vue/typescript`? I uninstalled `@vue/typescript` and installed `vue-tsc` instead and the reordering stopped working.

**Comment by mreduar** [comment]
> Published as `4.0.0` 🎉
> 
> Please try it out and let me know if there are any issues 🙏

For now it works well with `vue-tsc`
